src/web-worker/amp-worker.js

```diff
@@ -29,6 +29,38 @@ const TAG = 'web-worker';
 let PendingMessageDef;
 
 /**
+ * Error that can be passed through web worker
+ * @typedef {{
+ *   message: string,
+ *   stack: string,
+ * }}
+ */
+export let AmpWorkerErrorDef;
```

Comment:
IMO this typedef shouldn't live here since:

- It's not actually used by any code in this file and 
- It adds an unnatural dependency to `bind-evaluator` (the worker's use of bind-evaluator is an implementation detail, not vice-versa)

Move this to `bind-evaluator` since this only concerns the boundary between that class and `bind-impl`. Also, I think the helper methods below are simple enough to be unnecessary and not worthwhile as a layer of indirection.